src/controllers/legacycontrollersettings.cpp

```diff
@@ -11,6 +11,28 @@
 
 #include "moc_legacycontrollersettings.cpp"
 
+namespace {
+
+// Regex that allows to use :hwbtn:`LABELSTRING` in xml controller settings xml
+const QRegularExpression kHwbtnRe(QStringLiteral(":hwbtn:`([^`]*)`"));
+
+const QString kHwbtnStyleWrapper() {
+    return QStringLiteral(
+            "<span style='background-color: #111111; "
+            "color: #d9d9d9; "
+            "padding: 2px; "
+            "font-weight: 700;'>"
+            "\\1</span>");
+}
+
+QString replaceMarkupStyleStr(const QString& rawStr) {
+    QString str = rawStr;
+    str.replace(kHwbtnRe, kHwbtnStyleWrapper());
+    return str;
+}
```

Comment:
at that point, just take by value. also returning directly looks nicer.
```suggestion
QString replaceMarkupStyleStr(QString str) {
    return str.replace(kHwbtnRe, kHwbtnStyleWrapper());
}
```

Reply:
sure, I vaguely remembered there was a rule to avoid that

Reply:
oh, right. well I'm not sure actually. It returns a reference, so it might be dangling, but then that API wouldn't make sense at all... I'm pretty sure its fine because we're returning a value.